packages/protocol-parser/src/Schema.ts

```diff
@@ -0,0 +1,160 @@
+import {
+  StaticPrimitiveType,
+  DynamicPrimitiveType,
+  staticAbiTypeToByteLength,
+  DynamicAbiType,
+  StaticAbiType,
+  schemaAbiTypes,
+} from "@latticexyz/schema-type";
+import { Hex, hexToNumber, sliceHex } from "viem";
+import { decodeDynamicField } from "./decodeDynamicField";
+import { decodeStaticField } from "./decodeStaticField";
+import { hexToPackedCounter } from "./hexToPackedCounter";
+import { InvalidHexLengthForSchemaError, SchemaStaticLengthMismatchError } from "./errors";
+import { encodeFieldData } from "./encodeField";
+
+export class Schema {
+  readonly staticFields: readonly StaticAbiType[];
+  readonly dynamicFields: readonly DynamicAbiType[];
+
+  constructor(staticFields: readonly StaticAbiType[], dynamicFields: readonly DynamicAbiType[] = []) {
+    // TODO: validate at least one static field
```

Comment:
i believe there can also just be dynamic fields? are we even enforcing that there are any fields?

Reply:
oh I think you might be right, will remove